tests/test_statecontainers.py

```diff
@@ -25,6 +25,28 @@ def setUp(self):
                                                                    parentHandle='456')
         self.dc.DescriptorVersion = 42
 
+    def test_diff_float(self):
+        """Verify correct results of ContainerBase.diff() method for float values."""
+        sc1 = statecontainers.ClockStateContainer(nsmapper=self.nsmapper, descriptorContainer=self.dc)
+        sc2 = statecontainers.ClockStateContainer(nsmapper=self.nsmapper, descriptorContainer=self.dc)
+        sc1.LastSet = 0.0
+        sc2.LastSet = 0.0
+        # if sc1 is zero, a diff < 1e-6 is considered equal enough
+        self.assertEqual(0, len(sc1.diff(sc2)))
+        sc2.LastSet = 1e-7
+        self.assertEqual(0, len(sc1.diff(sc2, max_float_diff = 1e-6)))
+        sc2.LastSet = 1e-5
+        self.assertEqual(1, len(sc1.diff(sc2, max_float_diff = 1e-6)))
+
+        # if sc1 is not zero, the value of abs((sc1-sc2)/sc1) < 1e-6 is considered equal enough

```

Comment:
this statement is not correct anymore.
maybe the unittest should be rewritten to test that float values are evaluated as equal with a default precision of 15 digits.  as defined by math.isclose and sys.float_info.dig resulting in -> max( 1e-15 * max( abs(float1), abs(float2) ), 1e-15)

it maybe makes sense to test the diff()-method with 
1. two exactly equal floats having 15 digits, using the default value of max_float_diff
2. two time.time()-values where one was cut after the millisecond digit, using a suitable value for max_float_diff

Reply:
accepted as is